pkg/sql/schemachanger/doc.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package schemachanger
+
+// The schemachanger package and its descendents contain the declarative schema
+// changer. This machinery is the latest iteration of how CockroachDB performs
+// schema changes, in contrast with the so-called legacy schema changer. This
+// doc file provides a high-level overview of its architecture, the aim being to
+// provide a useful introduction to the code and the design decisions behind
+// it.
+//
+// # Similarities and differences with the legacy schema changer
+//
+// The main challenge that any schema changer in CockroachDB must solve is to
+// perform these correctly while maintaining the "online" property: tables must
+// remain accessible to concurrent queries while they undergo schema changes.
+// Online schema changes are enabled by upholding the 2-version invariant, which
```

Comment:
should we also briefly explain what his "2-version invariant" is?

Reply:
I'll add a link to the RFC.